app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/manager/AccessibilityFieldManager.kt

```diff
@@ -0,0 +1,30 @@
+package com.x8bit.bitwarden.data.autofill.accessibility.manager
+
+import android.net.Uri
+import android.view.accessibility.AccessibilityNodeInfo
+
+/**
+ * A manager for finding fields that match particular characteristics.
+ */
+interface AccessibilityFieldManager {
+    /**
+     * A helper function for retrieving the appropriate nodes based on the given [predicate].
+     *
+     * This function is recursive but will stop recurring if the [recursionDepth] reaches more that
+     * the [MAX_RECURSION_DEPTH].
```

Comment:
I see what we are doing here, I wonder if we kind of flip the logic a bit, where the function takes a `maxRecursionDepth` and goes backwards, and the contract owns the constant which we could pass a a default.

```
fun findAccessibilityNodeInfoList(
        rootNode: AccessibilityNodeInfo,
        maxDepth: Int = MAX_RECURSION_DEPTH,
        predicate: (AccessibilityNodeInfo) -> Boolean,
    ): List<AccessibilityNodeInfo>
    
    // the condition would if max depth <= 0 return
    // and when calling in the impl `maxDepth = maxDepth - 1`
```